### PR TITLE
feat: Add a JS Question

### PR DESCRIPTION
**Pull request by itsNue**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

Added a JS Question about Asynchronous Functions.

### CODE REVIEW

File: src/data/javascript-quiz.ts [hunk 1]
```diff
@@ -1772,7 +1772,18 @@ const javascriptQuiz = [
...
+  {
+    Question:
+      "Which one is a Native Function to run code asynchronously in JavaScript ?",
+    Answer: "setTimeOut()",
+    Distractor1: "repeat()",
+    Distractor2: "timeOut()",
+    Distractor3: "delay()",
+    Explanation:
+      "setTimeout() function will not pause execution of other functions in the functions stack, so it's an asynchronous function and it's a native function.",
+    Link: "https://developer.mozilla.org/en-US/docs/Web/API/setTimeout"
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1785]:
```suggestion
    Link: "https://www.freecodecamp.org/news/javascript-settimeout-how-to-set-a-timer-in-javascript-or-sleep-for-n-seconds/"
```

File: src/data/javascript-quiz.ts [hunk 2]
```diff
@@ -1772,7 +1772,18 @@ const javascriptQuiz = [
     Explanation:
       "The export keyword is used to make it possible to access small portions of code in other modules.",
     Link: "https://www.freecodecamp.org/news/javascript-modules-explained-with-examples/#"
+  },
+  {
+    Question:
+      "Which one is a Native Function to run code asynchronously in JavaScript ?",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1778]:
```suggestion
      "Which of the following options is a web API that runs code asynchronously in JavaScript?",
```

File: src/data/javascript-quiz.ts [hunk 3]
```diff
@@ -1772,7 +1772,18 @@ const javascriptQuiz = [
...
+    Answer: "setTimeOut()",
+    Distractor1: "repeat()",
+    Distractor2: "timeOut()",
+    Distractor3: "delay()",
+    Explanation:
+      "setTimeout() function will not pause execution of other functions in the functions stack, so it's an asynchronous function and it's a native function.",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1784]:
```suggestion
      "The setTimeout() function is an asynchronous function that will not pause execution of other functions in the stack.",
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

**Comment by itsNue** [comment]
I've commited the changes you've requested.
And thank you for taking your time to check this. Have a great day ahead!!

**Review by jdwilkin4 — APPROVED** [review]
Congrats on your first contribution to our repository. 🎉
We look forward to future contributions. 👍

**Pull request merged**